app/javascript/components/key-performance-indicators/utils/utc-date.js

```diff
@@ -0,0 +1,41 @@
+export default class UTCDate {
```

Comment:
We have the datefns lib in primero... Does it already do some of this?

Reply:
Yeah, I think I figured out that we don't need any of this anyway 😅. The issue turned out to be when displaying the dates, not when capturing them.

Reply:
Ok, I see why this is here now. All date math in js (including datefns afaik) is in the clients local timezone so when ever you want the beginning of a month you'll get e.g. `2021/08/01T00:00:00` and then when I convert that to UTC to send to the server it becomes `2021/07/31T23:00:00` which is the wrong month.

I'm reluctant to re-write it now at the 11th hour as this solution has been fairly rigorously put through QA but perhaps we could circle back to it with a datefns based solution? I too would prefer to use it.

You can peek at the commit that introduced this here: ca1e0f9ceb2be34e62eb78d7b59a6f7b0cb909b8 and a previous implementation here: 2a199d06c78ddda684cd13d667303e28fb1ba7e3